### ISSUE TITLE
Parsing Error in Pod Install

### ISSUE THREAD

**Issue by sibinlutoken** [CocoaPods/CocoaPods#7268]
# Report
if you include Google Grpc pod in your project, the pod install will fail

**pod '!ProtoCompiler'
pod '!ProtoCompiler-gRPCPlugin'**

Error:
_[!] ERROR: Parsing unable to continue due to parsing error:
contained in the file located at /Users/sibinlu/token-ios/sdk-objc/Podfile.lock
PODS:
  - !ProtoCompiler (3.4.0):
    - Protobuf (~> 3.0)
  - !ProtoCompiler-gRPCPlugin (1.7.1):
    - !ProtoCompiler (= 3.4.0)
    - gRPC-ProtoRPC (= 1.7.1)_

## What did you do?

**pod '!ProtoCompiler'
pod '!ProtoCompiler-gRPCPlugin'**
in the podfile

**pod update**
generate podfile.lock

**pod install**
the parsing error above is thrown 

## What did you expect to happen?

install and parse correctly

## What happened instead?

parsing error

## CocoaPods Environment

COCOAPODS: 1.4.0.beta.2
COCOAPODS: 1.3.1
are broken

COCOAPODS: 1.2.1
can bypass the problem

The reason of the parsing error is we shall have **'!ProtoCompiler'** instead of  **!ProtoCompiler** in the podfile.lock. Otherwise it will fail the parsing in pod install.

Please fix this. Version 1.40 shall include this fix.

**Comment by dnkoutso** [comment]
PR open by https://github.com/CocoaPods/Core/pull/419

**Comment by orta** [comment]
Duplicate of #6955

**Comment by combinatorial** [comment]
This is marked as a duplicate, yet the fix is merged https://github.com/CocoaPods/Core/pull/395

And this continues to be an issue in 1.4.0.rc.1, it would seem that this issue is not resolved by the referenced fix.

**Comment by phlippieb** [comment]
Agreed that this isn't fixed in 1.4.0.rc.1. Running `pod install` twice with a podfile that contains `pod '!ProtoCompiler'` results in the error
```
[!] ERROR: Parsing unable to continue due to parsing error:
contained in the file located at /path/to/Podfile.lock
```
